docs/src/api.ts

```diff
@@ -53,21 +58,59 @@ async function createError(response: Response): Promise<ApiError> {
 
 export async function renderImage(body: object): Promise<Blob> {
   const requestObject = buildRequestRenderImage(body);
+  return retryRequest('/render_image', requestObject);
+}
 
-  const backend_url: URL = new URL(
-    config.customFields.environment === 'development'
-      ? 'http://localhost:8081'
-      : 'https://playground.compositor.live'
-  );
-  const renderImageUrl = new URL('/render_image', backend_url);
-  let response;
-  try {
-    response = await fetch(renderImageUrl, requestObject);
-  } catch (error) {
-    throw new ApiError(error.message);
-  }
-  if (response.status >= 400) {
-    throw await createError(response);
+async function fetchWithTimeout(
+  path: string,
+  requestObject: RequestObject,
+  timeout: number
+): Promise<Response> {
+  const fullUrl = new URL(path, BACKEND_URL);
+
+  const controller = new AbortController();
+  const signal = controller.signal;
+
+  return new Promise<Response>((res, rej) => {
+    const timer = setTimeout(() => {
+      controller.abort();
+      rej(new ApiError(`Fetch not responding after ${timeout / 1000} s`));
+    }, timeout);
+
+    fetch(fullUrl, { ...requestObject, signal })
+      .then(response => {
+        clearTimeout(timer);
+        res(response);
+      })
+      .catch(error => {
+        clearTimeout(timer);
+        rej(new ApiError(error.message));
+      });
+  });
+}
+
+async function retryRequest(path, requestObject) {
```

Comment:
I wasn't sure what exactly should remain in `retryRequest` function, so for now I did it like this